eups.tex

```diff
@@ -2,19 +2,19 @@ \subsection{EUPS}\label{sec:eups}
+EUPS was initially developed at FNAL and adopted by SDSS \{citation needed\} as UPS, and has evolved over time migrating from C to Perl to Python.
+The EUPS philosophy is to organize the hierarchy of components that are required to make a release of a complex software product on a particular platform (\textit{e.g.} os/x); optional and required dependencies are declared recursively using ``table'' files. 
+As products are built they are installed into a directory tree where each built product is isolated from other products and associated with the exact set of dependencies used to build it.
+All built versions of a product are available at any time, along with the correct set of dependencies stored as ``expanded'' table files.
+When a user requests that a particular product be ``set up'', EUPS sets environment variables in the current shell identifying the location of the appropriate version of every dependency.
+Each product can specify which additional environment variables are relevant (for example \texttt{LD\_LIBRARY\_PATH}, \texttt{PATH}, \texttt{PYTHONPATH}) in the table file.
+For each product that is being set up, EUPS adjusts the environment and sets, prepends, or appends to any previous value as requested.
+When a product is ``un-set up'' this procedure is reversed and the environment cleaned up.
 
 A common use case is to set up a base environment and override versions of specific packages in order to debug some regression or to develop a new feature.
-EUPS excels at this use case since it is optmized for mixing and matching collections of products that have differing dependencies and ensuring that exactly the correct set are being tested.
-One downside to such a powerful system is that for many users they do not need to be able to mix and match versions and are happy to use whatever happens to be the most recent release of the software collection.
-For these users the EUPS user interface is far too complex and opaque and is much harder to understand than setting up a Python virtual environment and installing the relevant packages.
+EUPS excels at this use case since it is optimized for mixing and matching collections of products that have differing dependencies and ensuring that exactly the correct set are being tested; for example a user may use a specific version of a high-level product, with its exact dependencies, while overriding one or more dependencies with versions checked out from \textit{e.g.} \texttt{git}.
+EUPS supports the concept of tagging a `current' version, and many users are able to simply use this version; other options are to accept a `stable' or other well-known tag.
+Unfortunately LSST has not done a good job of educating our users in best practices, and some find the EUPS user interface to be far too complex and opaque and indeed harder to understand than setting up a Python virtual environment and installing the relevant packages.
```

Comment:
Maybe we say we've "not done a good job of educating our users in EUPS best practices"?